outline/shadowsocks/client.go

```diff
@@ -22,36 +27,70 @@ import (
 
 	"github.com/Jigsaw-Code/outline-go-tun2socks/outline"
 	"github.com/Jigsaw-Code/outline-go-tun2socks/outline/connectivity"
+	"github.com/Jigsaw-Code/outline-go-tun2socks/outline/internal/utf8"
 	"github.com/Jigsaw-Code/outline-go-tun2socks/outline/neterrors"
 	"github.com/Jigsaw-Code/outline-internal-sdk/transport"
 	"github.com/Jigsaw-Code/outline-internal-sdk/transport/shadowsocks"
 	"github.com/Jigsaw-Code/outline-internal-sdk/transport/shadowsocks/client"
 	"github.com/eycorsican/go-tun2socks/common/log"
 )
 
-// Config represents a shadowsocks server configuration.
-// Exported via gobind.
-type Config struct {
```

Comment:
We need to keep this for now, for backward compatibility. But mark it as deprecated.

Breaking the binary api will break nthlink, and will prevent us from incrementally upgrading the client to the new api.

Reply:
It's moved to `config.go`.